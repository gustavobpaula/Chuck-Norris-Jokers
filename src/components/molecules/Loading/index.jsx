import React from 'react';
import PropTypes from 'prop-types';
import loading from '../../../assets/images/loading.gif';
import { Image } from '../../atoms';

const Loading = ({ className, width, height, maxWidth, maxHeight }) => (
  <Image
    className={className}
    src={loading}
    alt="Loading"
    height={height}
    maxHeight={maxHeight}
    maxWidth={maxWidth}
    width={width}
  />
);

Loading.propTypes = {
  className: PropTypes.string,
  height: PropTypes.string,
  width: PropTypes.string,
  maxHeight: PropTypes.string,
  maxWidth: PropTypes.string,
};

Loading.defaultProps = {
  className: 'logo',
  maxHeight: '100px',
  maxWidth: '100%',
  height: 'auto',
  width: 'auto',
};

export default Loading;
