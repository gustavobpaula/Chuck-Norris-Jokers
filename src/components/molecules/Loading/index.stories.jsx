import React from 'react';
import Loading from './index';

export default {
  component: Loading,
  title: 'Molecules/Loading',
};

export const element = () => <Loading />;

element.story = {
  name: 'Default',
};

export const customSizes = () => <Loading width="50px" />;

customSizes.story = {
  name: 'Custom Sizes',
};
