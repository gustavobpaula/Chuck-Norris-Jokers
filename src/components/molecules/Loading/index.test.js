import React from 'react';
import { mount } from 'enzyme';
import Loading from './index';

describe('<Loading />', () => {
  it('should mount properly', async () => {
    const wrapper = mount(<Loading />);

    expect(wrapper).toMatchSnapshot();
  });
});
