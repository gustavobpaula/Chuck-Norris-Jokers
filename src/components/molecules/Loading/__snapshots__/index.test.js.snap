// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should mount properly 1`] = `
<Loading
  className="logo"
  height="auto"
  maxHeight="100px"
  maxWidth="100%"
  width="auto"
>
  <Image
    alt="Loading"
    className="logo"
    height="auto"
    maxHeight="100px"
    maxWidth="100%"
    src="test-file-stub"
    width="auto"
  >
    <style__Image
      alt="Loading"
      className="logo"
      height="auto"
      maxHeight="100px"
      maxWidth="100%"
      src="test-file-stub"
      width="auto"
    >
      <StyledComponent
        alt="Loading"
        className="logo"
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "style__Image-zsbil-0",
              "isStatic": false,
              "lastClassName": "nQWWf",
              "rules": Array [
                "height:",
                [Function],
                ";max-height:",
                [Function],
                ";max-width:",
                [Function],
                ";width:",
                [Function],
                ";",
              ],
            },
            "displayName": "style__Image",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "style__Image-zsbil-0",
            "target": "img",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        height="auto"
        maxHeight="100px"
        maxWidth="100%"
        src="test-file-stub"
        width="auto"
      >
        <img
          alt="Loading"
          className="style__Image-zsbil-0 nQWWf logo"
          height="auto"
          src="test-file-stub"
          width="auto"
        />
      </StyledComponent>
    </style__Image>
  </Image>
</Loading>
`;
